flink-core/src/main/java/org/apache/flink/api/connector/sink/Sink.java

```diff
@@ -93,6 +141,29 @@
         /** @return The id of task where the writer is. */
         int getSubtaskId();
 
+        /** @return number of parallel Sink tasks. */
+        int getNumberOfParallelSubtasks();
+
+        /** @return The metric group this writer belongs to. */
+        MetricGroup metricGroup();
+    }
+
+    /**
+     * The interface exposes some runtime info for creating a {@link Committer} or {@link
+     * GlobalCommitter}.
+     */
+    interface CommitterInitContext {
```

Comment:
I wonder whether we should only have one context to me there is no clear difference between the `InitContext` and the `CommitterContext` i.e. to support better-batching capabilities in the `CommiterContext` is probably also need access to the `InitContext#getProcessingTimeService`. 

I'd propose to keep everything in one context AFAIK it will be supplemented by the exact same fields from the runtime context. WDYT? @AHeise

Reply:
Mostly I wanted to avoid sharing `ProcessingTimeService` with committers. But maybe it's not a bad idea if you need to do some cleanup (hive small file compression?).

For global committer, I'm unsure how `getSubtaskId` and `getNumberOfParallelSubtasks` are defined. But we could just statically set 0/1.